protocols/imap/src/test/java/org/apache/james/imap/processor/base/SelectedMailboxImplTest.java

```diff
@@ -170,12 +183,138 @@ public void concurrentEventShouldBeProcessedSuccessfullyDuringInitialisation() t
         };
     }
 
-    private void emitEvent(MailboxListener mailboxListener) throws Exception {
+    void emitEvent(MailboxListener mailboxListener) throws Exception {
         mailboxListener.event(EventFactory.added()
             .randomEventId()
             .mailboxSession(MailboxSessionUtil.create(Username.of("user")))
             .mailbox(mailbox)
             .addMetaData(new MessageMetaData(EMITTED_EVENT_UID, MOD_SEQ, new Flags(), SIZE, new Date(), new DefaultMessageId()))
             .build());
     }
+
+    @Test
+    void updateApplicableFlagsShouldNotUpdateWhenEmptyFlagsUpdate() {
+        ApplicableFlags applicableFlags = ApplicableFlags.from(flagsBuilder().add(SEEN).build());
+        MailboxListener.FlagsUpdated flagsUpdated = flagsUpdated(updatedFlags().noOldFlag().noNewFlag());
+        ApplicableFlags actual = SelectedMailboxImpl.updateApplicableFlags(applicableFlags, flagsUpdated);
+        assertThat(actual).satisfies(ap -> {
+            assertThat(ap.updated()).isFalse();
+            assertThat(ap.flags()).isEqualTo(flagsBuilder().add(SEEN).build());
+        });
+    }
+
+    @Test
+    void updateApplicableFlagsShouldNotUpdateWhenNewFlag() {
+        ApplicableFlags applicableFlags = ApplicableFlags.from(flagsBuilder().add(SEEN).build());
+        MailboxListener.FlagsUpdated flagsUpdated =
+            flagsUpdated(updatedFlags().noOldFlag().newFlags(flags -> flags.add(ANSWERED)));
+        ApplicableFlags actual = SelectedMailboxImpl.updateApplicableFlags(applicableFlags, flagsUpdated);
+        assertThat(actual).satisfies(ap -> {
+            assertThat(ap.updated()).isFalse();
+            assertThat(ap.flags()).isEqualTo(flagsBuilder().add(SEEN).add(ANSWERED).build());
+        });
+    }
+
+    @Test
+    void updateApplicableFlagsShouldNotUpdateWhenSeveralUpdatedFlagsNewFlag() {
```

Comment:
I don't really understand when it should be updated.
What does the `ApplicableFlags.updated` really means ?
Could you include a test where updated is true ?

Reply:
the test title is messed up, sorry,
I didn't changed the logic, just try to cover with a testsuite.
AFAIU, you need a new UserFlag to switch updated to true (see https://github.com/linagora/james-project/pull/3447/files/f13a788229ed8307a1f500a3f92414c097b03ba2#diff-839d1c9f7d29102fd93fe81c80d27170R263 )

Reply:
Is there more explanation needed here?